### PR TITLE
Update to 0.16.1, Pytorch 2.1

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Review by carterbox — COMMENTED** [review]
Is CUDNN a direct dependency of torchvision? The logs show that it's an unused run export.

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=824944&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=841356e0-85bb-57d8-dbbc-852e683d1642&l=1114

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 # torchvision requires that CUDA major and minor versions match with pytorch
 # https://github.com/pytorch/vision/blob/fa99a5360fbcd1683311d57a76fcc0e7323a4c1e/torchvision/extension.py#L79C1-L85C1
 {% set torch_proc_type = "cuda" ~ cuda_compiler_version | replace('.', '') if cuda_compiler_version != "None" else "cpu" %}
 # Upstream has specific compatability ranges for pytorch and python which are
 # updated every 0.x release. https://github.com/pytorch/vision#installation
-{% set compatible_pytorch = "2.0" %}
+{% set compatible_pytorch = "2.1" %}
 
 package:
   name: torchvision-split
```
Comment (hmaarrfk) [inline recipe/meta.yaml:11]:
```suggestion
  name: torchvision
```

**Comment by hmaarrfk** [comment]
thank you for noticing that, but that hasn't change in this PR, it was also there before:
```
2023-11-13T08:40:45.2161860Z WARNING (torchvision): run-exports library package conda-forge::cudnn-8.8.0.121-h838ba91_3 in requirements/run but it is not used (i.e. it is overdepending or perhaps statically linked? If that is what you want then add it to `build/ignore_run_exports`)
```

**Comment by hmaarrfk** [comment]
I feel like the recipe simplification is out of scope. I reverted that change.

@carterbox while the `cudnn` is weird, it doesn't seem any different from before. I am inclined to merge unless there are other issues.

**Comment by carterbox** [comment]
I was just pointing it out. It doesn't seem to be a direct dependency of this package, but I didn't read through the git history of this recipe / upstream to know why it was added as a dependency in this recipe.

**Pull request merged**